Tests/SonarQube.TeamBuild.PreProcessor.Tests/SonarLintAnalyzerProviderTests.cs

```diff
@@ -0,0 +1,209 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SonarLintAnalyzerProviderTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.TeamBuild.Integration;
+using System.IO;
+using System.Linq;
+using TestUtilities;
+
+namespace SonarQube.TeamBuild.PreProcessor.Tests
+{
+    [TestClass]
+    public class SonarLintAnalyzerProviderTests
+    {
+        public TestContext TestContext { get; set; }
+
+        #region Tests
+
+        [TestMethod]
+        public void SonarLint_ValidRuleset()
+        {
+            // Arrange
+            string rootDir = CreateTestFolders();
+            TestLogger logger = new TestLogger();
+            TeamBuildSettings settings = CreateSettings(rootDir);
+            MockSonarQubeServer mockServer = CreateValidServer("valid.project", "valid.profile");
+
+            // Act
+            SonarLintAnalyzerProvider.SetupAnalyzers(mockServer, settings, "valid.project", logger);
+
+            // Assert
+            CheckRulesetExists(rootDir);
+            CheckBinariesExist(rootDir);
+        }
+
+        [TestMethod]
+        public void SonarLint_PluginNotInstalled()
+        {
+            // Arrange
+            string rootDir = CreateTestFolders();
+            TestLogger logger = new TestLogger();
+            TeamBuildSettings settings = CreateSettings(rootDir);
+
+            MockSonarQubeServer mockServer = CreateValidServer("valid.project", "valid.profile");
+            mockServer.Data.InstalledPlugins.Remove(SonarLintAnalyzerProvider.CSharpPluginKey);
```

Comment:
The setup for all of the "failure" tests is similar; they all start with an initially valid setup, then change something in the data model to make it invalid e.g. removing the C# plugin